src/main/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAO.java

```diff
@@ -67,15 +67,17 @@ public void deleteDataset(Dataset dataset, Integer userId) throws Exception {
   public void deleteStudy(Study study, User user) throws Exception {
     jdbi.useHandle(handle -> {
       handle.getConnection().setAutoCommit(false);
-      study.getDatasets().forEach(d -> {
-        try {
-          deleteDataset(d, user.getUserId());
-        } catch (Exception e) {
-          handle.rollback();
-          logException(e);
-          throw new DatasetDeletionException(e);
-        }
-      });
+      if (study.getDatasets() != null) {
+        study.getDatasets().forEach(d -> {
+          try {
+            deleteDataset(d, user.getUserId());
+          } catch (Exception e) {
+            handle.rollback();
+            logException(e);
+            throw new DatasetDeletionException(e);
```

Comment:
This throws `DatasetDeletionException` but nothing catches it and it's not specially handled in the top level exception handler, as far as I can tell. If you didn't need to throw a different exception here could put this inside the `try`/`catch` below by using a `for` loop instead of the `forEach()` call.

Reply:
It bubbles up to the Resource class where eventually it gets rethrown as a 500 to the client. I ended up not touching this after using Shelby's suggestion, so will leave this as is for now.